Revit_Core_Engine/Query/OrientationAngle.cs

```diff
@@ -58,31 +57,38 @@ public static double OrientationAngle(this FamilyInstance familyInstance, RevitS
         [Description("Get the orientation angle of a duct.")]
         [Input("Autodesk.Revit.DB.Mechanical.Duct", "Revit duct.")]
         [Input("BH.oM.Adapters.Revit.Settings.RevitSettings", "Revit settings.")]
-        [Output("double", "Orientation angle of a duct.")]
+        [Output("double", "Orientation angle of a duct in radians.")]
         public static double OrientationAngle(this Duct duct, RevitSettings settings = null)
         {
             settings = settings.DefaultIfNull();
+            Options options = new Options();
+            options.IncludeNonVisibleObjects = false;
 
             double rotation;
+
             Location location = duct.Location;
 
             LocationCurve locationCurve = location as LocationCurve;
-
             Curve curve = locationCurve.Curve;
 
-            XYZ startPoint = curve.GetEndPoint(0); // Start point
-            XYZ endPoint = curve.GetEndPoint(1); // End point
-
-            XYZ vector = (endPoint - startPoint).Normalize(); // Normalised vector
-
-            Transform transform = Transform.CreateTranslation(vector); // Transform of a vector
-
             BH.oM.Geometry.ICurve bhomCurve = curve.IFromRevit(); // Convert to a BHoM curve
 
+            // Get the duct connector
+            Connector connector = null;
+            foreach (Connector conn in duct.ConnectorManager.Connectors)
+            {
+                connector = conn;
+                break;
+            }
```

Comment:
As above, this should be an end connector, not any connector.

Reply:
Revised to find the right connector